src/components/pages/Onboarding/InviteVideoPreview.tsx

```diff
@@ -0,0 +1,243 @@
+import { RahaState } from "../../../store";
+import { connect, MapStateToProps } from "react-redux";
+import { getPrivateVideoInviteRef } from "../../../store/selectors/authentication";
+
+const BYTES_PER_MIB = 1024 * 1024;
+const MAX_MB = 60;
+const MAX_VIDEO_SIZE = MAX_MB * BYTES_PER_MIB;
+
+type ReduxStateProps = {
+  videoUploadRef?: firebase.storage.Reference;
+};
+
+type OwnProps = {
+  navigation: any;
+  invitingMember: Member;
+};
+
+type InviteVideoPreviewProps = ReduxStateProps & OwnProps;
+
+type InviteVideoStateProps = {
+  errorMessage?: string;
+  uploadStatus: UploadStatus;
+  inviteStatus: InviteStatus;
+  uploadedBytes: number;
+  totalBytes: number;
+};
+
+enum UploadStatus {
+  NOT_STARTED,
+  UPLOADING,
+  UPLOADED
+}
+
+enum InviteStatus {
+  NOT_REQUESTED,
+  REQUESTING,
+  REQUESTED
+}
+
+class InviteVideoPreviewView extends React.Component<
+  InviteVideoPreviewProps,
+  InviteVideoStateProps
+> {
+  videoUri?: string;
+
+  constructor(props: InviteVideoPreviewProps) {
+    super(props);
+    this.state = {
+      uploadStatus: UploadStatus.NOT_STARTED,
+      uploadedBytes: 0,
+      totalBytes: 0,
+      inviteStatus: InviteStatus.NOT_REQUESTED
+    };
+  }
+
+  navigateToCamera() {
+    this.props.navigation.navigate(RouteName.OnboardingCamera, {
+      invitingMember: this.props.navigation.getParam("invitingMember", null),
+      verifiedName: this.props.navigation.getParam("verifiedName", null)
+    });
+  }
+
+  uploadVideo = async (videoUploadRef: firebase.storage.Reference) => {
+    const videoUri = this.props.navigation.getParam("videoUri", null);
+    this.setState({ uploadStatus: UploadStatus.UPLOADING });
+    const response = await fetch(videoUri);
+    const blob = await response.blob();
+    //@ts-ignore Expo Blob does not have data type
+    if (blob.data.size <= MAX_VIDEO_SIZE) {
+      // TODO: Transcode video to make it smaller.
+      const metadata = {
+        //@ts-ignore Expo Blob does not have data type
+        contentType: blob.data.type
+      };
+      const uploadTask = videoUploadRef.put(blob, metadata);
+      uploadTask.on(
+        firebase.storage.TaskEvent.STATE_CHANGED,
+        s => {
+          const snapshot = s as firebase.storage.UploadTaskSnapshot;
+          this.setState({
+            uploadedBytes: snapshot.bytesTransferred,
+            totalBytes: snapshot.totalBytes
+          });
+        },
+        () =>
+          this.setState({
+            errorMessage: "Could not upload. Please try again.",
+            uploadStatus: UploadStatus.NOT_STARTED
+          }),
+        () => {
+          this.setState({ uploadStatus: UploadStatus.UPLOADED });
+          this.requestInvite();
+        }
+      );
+    } else {
+      this.setState({
+        errorMessage:
+          "The video size is larger than " +
+          MAX_VIDEO_SIZE +
+          "MB. Please retake your video."
+      });
+    }
+  };
+
+  requestInvite() {
+    this.setState({
```

Comment:
I think you can use @rahulgi ’s code for `Give` as a model for doing this, ask him about it